common/src/chain/transaction/signature/inputsig/authorize_pubkey_spend.rs

```diff
@@ -72,200 +72,107 @@ mod test {
     use crate::{
         address::pubkeyhash::PublicKeyHash,
         chain::{
-            signature::{
-                inputsig::{InputWitness, StandardInputSignature},
-                sighashtype::SigHashType,
-                signature_hash,
-            },
-            Destination, Transaction, TransactionCreationError, TxInput, TxOutput,
+            signature::{inputsig::StandardInputSignature, signature_hash},
+            transaction::signature::tests::utils::{generate_unsigned_tx, sig_hash_types},
+            Destination,
         },
-        primitives::{Amount, Id},
     };
-    use crypto::key::{KeyKind, PrivateKey, PublicKey};
+    use crypto::key::{KeyKind, PrivateKey};
 
-    fn generate_unsigned_tx(
-        outpoint_dest: Destination,
-    ) -> Result<Transaction, TransactionCreationError> {
-        let tx = Transaction::new(
-            0,
-            vec![TxInput::new(
-                Id::<Transaction>::new(&H256::zero()).into(),
-                0,
-                InputWitness::NoSignature(None),
-            )],
-            vec![TxOutput::new(Amount::from_atoms(100), outpoint_dest)],
-            0,
-        )?;
-        Ok(tx)
-    }
-
-    fn make_data_for_verify(
-        sighash_type: SigHashType,
-    ) -> (PublicKey, AuthorizedPublicKeySpend, H256) {
-        const INPUT_NUM: usize = 0;
-        let (private_key, public_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
-        let outpoint_dest = Destination::PublicKey(public_key.clone());
-        let tx = generate_unsigned_tx(outpoint_dest.clone()).unwrap();
-        let witness = StandardInputSignature::produce_signature_for_input(
-            &private_key,
-            sighash_type,
-            outpoint_dest,
-            &tx,
-            INPUT_NUM,
-        )
-        .unwrap();
-        let spender_signature =
-            AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap();
-        let sighash = signature_hash(witness.sighash_type(), &tx, INPUT_NUM).unwrap();
-        (public_key, spender_signature, sighash)
-    }
+    const INPUT_NUM: usize = 0;
```

Comment:
Would be better if this number can be a random one. I don't like constants in the tests.

Reply:
Btw, a simple function that generates a random number is sufficient for this.